scripts/AbstractWidget.py

```diff
@@ -0,0 +1,322 @@
+import re
+import sys
+import os
+class eventHandlerInfo:
+    def __init__(self, java_src, identity, java_line_number,xml_line_number = None, xml_src = None, name=None):
+        self.xml_src = xml_src
+        self.java_src = java_src
+        self.identity = identity
+        self.java_line_number = java_line_number
+        self.xml_line_number = xml_line_number
+        self.name = name
+
+    def toString(self):
+        result = []
+        result.append("Java Source: " + self.java_src + ":" + self.java_line_number)
+        if self.xml_src != None:
+            result.append("XML source:" + self.xml_src + ":" + str(self.xml_line_number))
+        return result
+
+    def getName(self):
+        return self.name
+
+class dialogUI:
+    def __init__(self, node):
+        self.node = node
+        self.message = None
+        self.title = None
+        self.positive = None
+        self.negative = None
+        self.neutral = None
+
+    def findNode(self, root, targetClass, targetId):
+        for item in root.children:
+            if item.node_class == targetClass and item.res_id == targetId:
+                return item
+        return None
+
+    def isDialogLL(self):
+        LinearLayout = self.findNode(self.node, "LinearLayout", "parentPanel")
+        if not LinearLayout:
+            return False
+        FrameLayout = self.findNode(LinearLayout, "FrameLayout", "contentPanel")
+        if not FrameLayout:
+            return False
+        ScrollView = self.findNode(FrameLayout, "ScrollView", "scrollView")
+        if not ScrollView:
+            return False
+        __LinearLayout = self.findNode(ScrollView, "LinearLayout", None)
+        if not __LinearLayout:
+            return False
+        TextView = self.findNode(__LinearLayout, "TextView", "message")
+        if not TextView:
+            return False
+        self.message = TextView.text
+
+        '''
+        TODO:
+        subject to change, there may be title and no message
+        '''
+
+        ButtonPanel = self.findNode(LinearLayout, "ScrollView", "buttonPanel")
+        if not ButtonPanel:
+            return False
+        __LinearLayout = self.findNode(ButtonPanel, "LinearLayout", None)
+        if not __LinearLayout:
+            return False
+
+        button1, button2, button3 = None, None, None
+        for button in LinearLayout.children:
+            if "button1" in button.value_str:
+                button1 = button.text
+            elif "button2" in button.value_str:
+                button2 = button.text
+            elif "button3" in button.value_str:
+                button3 = button.text
+
+        self.button1 = button1
+        self.button2 = button2
+        self.button3 = button3
+        return True
+    
+
+    def isDialogLLC(self):
```

Comment:
This name is misleading since `isXXX` is typically a pure computation that decides whether something is true, **without** changing any state.